Makefile

```diff
@@ -0,0 +1,496 @@
+	fi
+	@echo "NDK configured at $(ANDROID_NDK_ROOT)"
+	@echo "  toolchain: $(NDK_TOOLCHAIN)"
+	@echo "  clang-rt:  $(CLANG_RT_DIR)"
+
+
+$(STAMPS)/android-armv7: android-env $(STAMPS)
+	$(CARGO) rustc --locked \
+	  --target armv7-linux-androideabi \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: android-armv7
+android-armv7: $(STAMPS)/android-armv7
+
+$(STAMPS)/android-armv8: android-env $(STAMPS)
+	$(CARGO) rustc --locked \
+	  --target aarch64-linux-android \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: android-armv8
+android-armv8: $(STAMPS)/android-armv8
+
+$(STAMPS)/android-x86: android-env $(STAMPS)
+	# Link clang_rt.builtins statically for x86_64 Android
+	$(CARGO) rustc --locked \
+	  --target x86_64-linux-android \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- \
+	  -C strip=symbols \
+	  -l static=clang_rt.builtins-x86_64-android \
+	  -L $$(dirname $$($(ANDROID_NDK_HOME)/toolchains/llvm/prebuilt/$(PLATFORM_DIR)/bin/clang --print-runtime-dir))/linux
+	@touch $@
+
+.PHONY: android-x86
+android-x86: $(STAMPS)/android-x86
+
+$(STAMPS)/android: $(STAMPS)/android-armv7 $(STAMPS)/android-armv8 $(STAMPS)/android-x86
+	@touch $@
+
+.PHONY: android
+android: $(STAMPS)/android
+
+.PHONY: android-test $(STAMPS)
+android-test: android
+	cd crypto-ffi/bindings && \
+	./gradlew android:build -x lint -x lintRelease
+
+#-------------------------------------------------------------------------------
+# 8) JVM native builds (Darwin + Linux)
+#-------------------------------------------------------------------------------
+
+# darwin build
+$(STAMPS)/jvm-darwin: $(STAMPS)/bindings-kotlin-jvm $(STAMPS)
+	cd crypto-ffi && \
+	$(CARGO) rustc --locked \
+	  --target aarch64-apple-darwin \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: jvm-darwin
+jvm-darwin: $(STAMPS)/jvm-darwin
+
+# linux build
+$(STAMPS)/jvm-linux: $(STAMPS)/bindings-kotlin-jvm $(STAMPS)
+	cd crypto-ffi && \
+	$(CARGO) rustc --locked \
+	  --target x86_64-unknown-linux-gnu \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: jvm-linux
+jvm-linux: $(STAMPS)/jvm-linux
+
+.PHONY: jvm
+ifeq ($(UNAME_S),Linux)
+jvm: jvm-linux
+else ifeq ($(UNAME_S),Darwin)
+jvm: jvm-darwin
+else
+$(error Unsupported host platform for jvm: $(UNAME_S))
+endif
+
+.PHONY: jvm-test
+jvm-test: jvm
+	cd crypto-ffi/bindings && \
+	./gradlew jvm:build -x lint -x lintRelease
+
+
+#-------------------------------------------------------------------------------
+# 9) Aggregate targets
+#-------------------------------------------------------------------------------
+
+.PHONY: mobile wasm all
+mobile: android ios
```

Comment:
We should also skip `ios` here if we're not on Darwin, just like in the other place.